### PR TITLE
feat: FR-1932 - added captcha for login/sign up; removed unused compo…

### PR DESCRIPTION
**Pull request by AntonReact**
…nents demosaas

### CODE REVIEW

File: packages/auth/src/components/ReCaptcha.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+interface IReCaptchaProps {
+  callback(token: string): void;
+  action: string;
+}
+
+const stateMapper = ({ captchaState }: AuthState) => captchaState;
+
+export const ReCaptcha: FC<IReCaptchaProps> = ({ callback, action }) => {
+  const state = useAuth(stateMapper);
```
Comment (serslon) [inline packages/auth/src/components/ReCaptcha.tsx:14]: Use here the `useField` hook and remove changes the value from the components to this component.
In this case, we can use the `useCallback` hook for setting a value to the formik.

**Review by serslon — APPROVED** [review]

File: packages/rest-api/src/auth/interfaces.ts [hunk 2]
```diff
@@ -128,6 +133,7 @@ export interface IVendorConfig {
 export interface ISignUpUser {
   email: string;
   companyName: string;
+  recaptchaToken: string;
```
Comment (frontegg-david) [inline packages/rest-api/src/auth/interfaces.ts:136]: add it as optional

File: packages/rest-api/src/auth/interfaces.ts [hunk 3]
```diff
@@ -6,7 +6,12 @@ export type IPostLogin = {
   SAMLResponse: string;
 };
 
-export type ILogin = { email: string; password: string };
+export interface ICaptchaPolicy {
+  siteKey: string;
+  enabled: boolean;
+}
+
+export type ILogin = { email: string; password: string; recaptchaToken: string };
```
Comment (frontegg-david) [inline packages/rest-api/src/auth/interfaces.ts:14]: add recaptchaToken as optional

**Review by frontegg-david — CHANGES_REQUESTED** [review]

**Comment by frontegg-david** [comment]
@AntonReact 
see request changes

**Review by frontegg-david — APPROVED** [review]

**Pull request merged**